### PR TITLE
feat: change java feature gate to cli flag

### PR DESCRIPTION
**Pull request by dexter0195**
**Description:** 
Change java instrumentation into command line

**Link to tracking Issue(s):**
https://github.com/open-telemetry/opentelemetry-operator/issues/2673

- Resolves: #2673

**Testing:**

**Documentation:**

### CODE REVIEW

**Comment by jaronoff97** [comment]
@dexter0195 looks like some unit test failures for the podmutator_test.go

**Review by swiatekm-sumo — COMMENTED** [review]
Looks like you have some unit tests failing because the new option is not enabled by default in the config constructor. Simplest way to fix these failures would be to enable it by default in `config.New`.

**Comment by dexter0195** [comment]
Hey @swiatekm-sumo thank you for the tip ! 😃 
I set the java intrumentation flag to default true in the config as you said and it worked ! I also needed to add a call to the config contructor in the java tests, see commit https://github.com/open-telemetry/opentelemetry-operator/pull/2831/commits/776ffb674bb8d9c724d1a61cc1a07c259066add8 

Please let me know what you think 🙏

**Review by swiatekm-sumo — APPROVED** [review]
LGTM, thanks for doing this!

File: internal/config/main.go [hunk 1]
```diff
@@ -74,6 +75,7 @@ func New(opts ...Option) Config {
 		operatorOpAMPBridgeConfigMapEntry: defaultOperatorOpAMPBridgeConfigMapEntry,
 		logger:                            logf.Log.WithName("config"),
 		version:                           version.Get(),
+		enableJavaInstrumentation:         true,
```
Comment (jaronoff97) [inline internal/config/main.go:78]: is there a reason we need to set this here if our default for the option is true?

**Review by iblancasa — APPROVED** [review]

**Pull request merged**